controllers/keystoneapi_controller.go

```diff
@@ -1290,6 +1290,17 @@ func (r *KeystoneAPIReconciler) generateServiceConfigMaps(
 	templateParameters["KeystoneEndpointPublic"], _ = instance.GetEndpoint(endpoint.EndpointPublic)
 	templateParameters["KeystoneEndpointInternal"], _ = instance.GetEndpoint(endpoint.EndpointInternal)
 
+	// MTLS params
+	if mc.Status.MTLSCert != "" {
+		templateParameters["MemcachedAuthCert"] = fmt.Sprint(memcachedv1.CertMountPath())
+		templateParameters["MemcachedAuthKey"] = fmt.Sprint(memcachedv1.KeyMountPath())
+		templateParameters["MemcachedAuthCa"] = fmt.Sprint(memcachedv1.CaMountPath())
+	} else {
+		templateParameters["MemcachedAuthCert"] = ""
+		templateParameters["MemcachedAuthKey"] = ""
+		templateParameters["MemcachedAuthCa"] = ""
```

Comment:
Could this break the current configuration? If I understand this correctly, it removes the TLS cert, key and ca, if the mTLS cert is not provided, which currently, even with the TLS Everywhere enabled configuration, wouldn't be present.

Reply:
as far as I know the current TLS-e keystone.conf is like this:

`[cache]

backend = dogpile.cache.pymemcache
memcache_servers=memcached-0.memcached.openstack.svc:11212,memcached-1.memcached.openstack.svc:11212,memcached-2.memcached.openstack.svc:11212
enable_retry_client = true
retry_attempts = 2
retry_delay = 0

enabled=true
tls_enabled=true`

and the CA certificates used are the ones under /etc/pki/tls etc.

This patch is only additive as it adds to the [cache] sections are as per https://docs.openstack.org/keystone/latest/configuration/samples/keystone-conf.html to perform the client auth if required.
I hope I am not missing anything.

Reply:
I think we could remove the else when we do `{{ if (index . "MemcachedAuthCert") }}` in the template as the condition.